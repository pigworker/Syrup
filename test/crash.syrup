one(<Bit>) -> @<Bit>
one(X) = Z where

  Z = dff(Q)
  Q = nand(X,Y)
  Y = Z

two(<Bit>) -> @<Bit>
two(X) = Z where

  Z = dff(Q)
  Q = nand(X,Y)
  Y = dff(Q)

experiment one
experiment two

experiment one = two