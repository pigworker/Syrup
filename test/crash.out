Circuits one and two are defined.

Truth table for one:
X { Z -> Z } Z
--{--------}--
0 { 0 -> 1 } 0
  { 1 -> 1 } 1
1 { 0 -> 1 } 0
  { 1 -> 0 } 1

Truth table for two:
X { Z Y -> Z Y } Z
--{------------}--
0 { 0 0 -> 1 1 } 0
  { 0 1 -> 1 1 } 0
  { 1 0 -> 1 1 } 1
  { 1 1 -> 1 1 } 1
1 { 0 0 -> 1 1 } 0
  { 0 1 -> 0 0 } 0
  { 1 0 -> 1 1 } 1
  { 1 1 -> 0 0 } 1

Bisimulation between one, two:
two can be distinguished from all possible states of one
when two has memory {10}
if one has memory like {1}, try inputs 0;1;0
if one has memory like {0}, try inputs 0

Bisimulation between two, one:
two can be distinguished from all possible states of one
when two has memory {10}
if one has memory like {1}, try inputs 0;1;0
if one has memory like {0}, try inputs 0
